### PR TITLE
Fix issue #76

### PR DESCRIPTION
**Pull request by bolt12**
Ignore effectStatusWrites in order to not generate bad schedules that lead to assertion failures.

Also fix issue introduced in 669d6b23c41d22f503916b7a924a11c39b442347 where the onlyReadEffect is not checking for effectStatusRead effects.

Check #76 for more details

### CODE REVIEW

File: io-sim/src/Control/Monad/IOSimPOR/Types.hs [hunk 1]
```diff
@@ -64,8 +64,17 @@ statusWriteEffects tids = mempty{effectStatusWrites = tids}
 someTvarId :: SomeTVar s -> TVarId
 someTvarId (SomeTVar r) = tvarId r
 
+-- | Make sure we only have read effects
+--
+-- It seems we need to ignore statusWrites effects to avoid generating bad
+-- schedules. See issue #76 for examples.
```
Comment (coot) [inline io-sim/src/Control/Monad/IOSimPOR/Types.hs:70]: We must ignore the `statusWrite` effects.   This function is used to only limit the blocking steps which are blocked on `STM`.

Maybe this function should be written as:
```
onlyReadEffect e = effectWrite e == mempty
```
but I am not 100% sure.

Comment (coot) [reply io-sim/src/Control/Monad/IOSimPOR/Types.hs:70]: If you update the comment we can merge it.

**Review by coot — APPROVED** [review]
LGTM

File: io-sim/src/Control/Monad/IOSimPOR/Types.hs (hunk 1 above)
Comment (coot) [reply io-sim/src/Control/Monad/IOSimPOR/Types.hs:70]: @bolt12 will the `onlyReadEffect` that I suggest above fix the test case which you mentioned here: https://github.com/input-output-hk/io-sim/issues/76#issuecomment-1495946252

Comment (bolt12) [reply io-sim/src/Control/Monad/IOSimPOR/Types.hs:70]: The fix I have fixes the test case mentioned in #76 . I am going to try your suggestion

File: io-sim/src/Control/Monad/IOSimPOR/Types.hs [hunk 2]
```diff
@@ -82,9 +82,11 @@ racingEffects e e' =
...
    || effectWrites e       `intersects`  effectReads  e'
    || effectWrites e       `intersects`  effectWrites e'
-   || effectStatusReads e  `intersectsL` effectStatusWrites e'
-   || effectStatusWrites e `intersectsL` effectStatusReads  e'
-   || effectStatusWrites e `intersectsL` effectStatusWrites e'
+   {--
+     - || effectStatusReads e  `intersectsL` effectStatusWrites e'
+     - || effectStatusWrites e `intersectsL` effectStatusReads  e'
+     - || effectStatusWrites e `intersectsL` effectStatusWrites e'
+     --}
```
Comment (bolt12) [inline io-sim/src/Control/Monad/IOSimPOR/Types.hs:89]: Should we leave a TODO and corresponding issue to get back to this?

**Review by bolt12 — COMMENTED** [review]
LGTM!

**Comment by coot** [comment]
Closing in favour of removal of `threadStatus`.

**Pull request closed without merging**